modules/concurrency/src/spinner.cpp

```diff
@@ -48,8 +48,14 @@ Spinner::~Spinner() {
 void Spinner::start() {
   throwExceptionIf<InvalidOperationException>(is_started_.load(), "Spinner is already started.");
 
-  // NOTE: Replace with std::stop_token and std::jthread when clang supports it.
-  spinner_thread_ = std::thread([this]() { spin(); });
+  async_spinner_handle_ = std::async(std::launch::async, [this]() mutable {
+    try {
+      spin();
+      this->exception_thrown_in_spin_promise_.set_value();
+    } catch (const std::exception& e) {
+      this->exception_thrown_in_spin_promise_.set_exception(std::current_exception());
+    }
```

Comment:
you do not need this, the exception is automatically captured and returned in the future. 
When you call `async_spinner_handle_.get()` if an exception was raised inside, it will be raised again